frontend/www/js/omegaup/login/reset.ts

```diff
@@ -0,0 +1,47 @@
+import { OmegaUp } from '../omegaup';
+import { types } from '../api_types';
+import * as api from '../api';
+import * as UI from '../ui';
+import T from '../lang';
+import Vue from 'vue';
+import login_PasswordReset from '../components/login/PasswordReset.vue';
+
+OmegaUp.on('ready', () => {
+  const payload = JSON.parse(
+    (<HTMLElement>document.getElementById('payload')).innerText,
+  );
+
+  let loginPaswwordRecover = new Vue({
+    el: '#main-container',
+    render: function(createElement) {
+      return createElement('omegaup-login-password-reset', {
+        props: {
+          email: payload.email,
+          resetToken: payload.resetToken,
+        },
+        on: {
+          'reset-password': function(
+            email: string,
+            resetToken: string,
+            password: string,
+            passwordConfirmation: string,
+          ) {
```

Comment:
siempre que se pueda, hay que usar [arrow functions](https://www.typescriptlang.org/docs/handbook/functions.html#this-and-arrow-functions) para evitar el rebinding de `this`.

```suggestion
          'reset-password': (
            email: string,
            resetToken: string,
            password: string,
            passwordConfirmation: string,
          ) => {
```

igual en L35